server/protocols/webadmin/webadmin-data-jmap/src/main/java/org/apache/james/webadmin/data/jmap/MessageFastViewProjectionCorrector.java

```diff
@@ -111,8 +115,8 @@ boolean failed() {
     private Mono<Void> correctUsersProjectionItems(Progress progress, Username username) {
         try {
             MailboxSession session = mailboxManager.createSystemSession(username);
-            return Flux.fromIterable(mailboxManager.search(MailboxQuery.privateMailboxesBuilder(session).build(), session))
-                .concatMap(mailboxMetadata -> Mono.fromCallable(() -> mailboxManager.getMailbox(mailboxMetadata.getId(), session)))
+            return listUsersMailboxes(session)
+                .concatMap(mailboxMetadata -> retrieveMailbox(session, mailboxMetadata))
```

Comment:
Why not  `flatMap`  here ?

Reply:
Because we don t want task paralelism to affect foreground.

See previous comments from @mbaecler